DXRFixes/DeusEx/Classes/Carcass.uc

```diff
@@ -316,10 +312,43 @@ function bool TryLootItem(DeusExPlayer player, Inventory item)
         }
     }
 
-    if (item.IsA('DeusExAmmo'))
+    if (item.IsA('Ammo'))
     {
-        if (DeusExAmmo(item).AmmoAmount == 0)
+        if (!dropsAmmo || Ammo(item).AmmoAmount == 0) {
+            DeleteInventory(item);
+            item.Destroy();
             return false;
+        }
+
+        playerAmmo = Ammo(player.FindInventoryType(item.class));
+
+        ammoAdded = Min( Ammo(item).AmmoAmount, Ammo(item).MaxAmmo - playerAmmo.AmmoAmount );
+        ammoLeftover = Ammo(item).AmmoAmount - ammoAdded;
+
+        if (ammoLeftover > 0) {
+            newAmmo = Spawn(Ammo(item).class,,, Location);
+            newAmmo.AmmoAmount = ammoLeftover;
+            newAmmo.Velocity = Velocity + VRand() * 280; // same as vanilla corpse drops
+        }
+
+        if (playerAmmo == None) {
+            TryLootRegularItem(player, item);
+        } else {
+            playerAmmo.AmmoAmount += ammoAdded;
```

Comment:
```suggestion
            playerAmmo.AddAmmo(ammoAdded);
```

Reply:
It's cleaner - if we ever had some reason that we needed to add code to AddAmmo, it would (hopefully) behave consistently across all cases where you get extra ammo.

Reply:
yea I know but it keeps it consistent with other code, and is safer to keep the sanity checks, and it's possible for a subclass to override the function (I don't see any in vanilla code but it's possible another mod might)